File: home-helper-core/src/main/kotlin/home/helper/core/domain/model/help_item/HelpItem.kt
/*
 * Copyright (c) 2022 Ryo-Kgym.
 */

package home.helper.core.domain.model.help_item

import home.helper.core.domain.model.help_point.HelpPoint

/**
 * お手伝い項目を保持するクラスです。
 */
data class HelpItem(
    val helpItemId: HelpItemId,
    val name: String,
    val helpPoint: HelpPoint,
    val memo: String? = null,
) {

    constructor(id: String,
                name: String,
                point: Int,
                memo: String? = null
    ) :
            this(
                helpItemId = HelpItemId(id),
                name = name,
                helpPoint = HelpPoint(point),
                memo = memo,
            )
}

File: home-helper-front/src/graphql/apollo/helpItem.ts
import { HelpItem } from "@domain/model/home_helper/HelpItem";
import { gql } from "@apollo/client";
import { query } from "@graphql/apollo/apollo-client";

export async function helpItems(): Promise<HelpItem[]> {
  const param = {
    query: GET_HELP_ITEMS,
    key: "helpItems",
  };

  return query(param);
}

const GET_HELP_ITEMS = gql`
  query getHelpItems {
    helpItems(param: {}) {
      id
      name
      point
      memo
    }
  }
`;
